### PR TITLE
make primitive string names case and underscore insensitive

### PR DESCRIPTION
**Pull request by tuethan1999**
### Pull Request Description
#1761 

### Feature Request
Provide convenience to a user by allowing them to pass a primitive name in either snake_case, TitleCase or camelCase

### Implementation
make the string to primitive matching case and underscore insensitive. Therefore `num_true`, `NumTrue`, `numTrue` and even `nU_mT_ru_e` all boil down to `numtrue`.

### CODE REVIEW

**Review by davesque — COMMENTED** [review]
Just want to plug my concerns again that we're enabling too broad a range of behaviors with this kind of functionality (previous discussion for reference: #1733).  But thanks all the same @tuethan1999 for getting it done. :)

Just had the one comment about a possible change.

May as well provide a link to the previous discussion for reference.

File: featuretools/tests/synthesis/test_dfs_method.py [hunk 1]
```diff
@@ -339,6 +339,29 @@ def test_no_warns_with_different_case_string_primitives(es):
         assert not record
 
 
+def test_no_warns_with_and_without_snakecase_string_primitives(es):
+    # Should not raise a UnusedPrimitiveWarning warnings
+    for trans_primitive in ["is_null", "isNull", "IsNull", "_iSnu__Ll_", "i_s_n_u_L_l"]:
```
Comment (thehomebrewnerd) [inline featuretools/tests/synthesis/test_dfs_method.py:344]: Am I understanding this correctly that we will accept `i_s_n_u_l_l` as a valid primitive name with this change? 

If so, is that what we intend here? That seems a bit too flexible to me.

Comment (thehomebrewnerd) [reply featuretools/tests/synthesis/test_dfs_method.py:344]: If we don't want to break the current behavior, I think we need to allow `is_null`, but beyond that, it seems like the only other things we should match on is the exact value returned by `primitive.__name__` (`IsNull`). I'm not even sure we should match on camel case, but that just my two cents here.

**Comment by tuethan1999** [comment]
I think we shouldn't try to support both `primitive.name` and `primitive.__name__` for the case where the customizable `primitive.name` conflicts with another primitive's `primitive.__name__`. Consider the case:

```python
class NumUnique():
  name = "foobar"

class UniqueCount():
  name = "num_unique"
```

**Pull request closed without merging**